fishtank/src/backend/docker.ts

```diff
@@ -85,4 +85,20 @@ export class Docker {
     }
     await this.cmd(runArgs, {})
   }
+
+  async createNetwork(
+    name: string,
+    options?: { driver?: string; attachable?: boolean; internal?: boolean },
+  ): Promise<void> {
```

Comment:
Is this response type defined or you are going to change it soon?

Reply:
You mean the return value? `Promise<void>`? I think this is what is going to be for the future. I do not see any reason to return anything else other than `void`, but who knows what we will need in the future... time will tell. By reading the specification document, do you see any use case for any return value other than `void`?

Reply:
In a testing flow, when a user created a network, how to interact with this network object again? by name? If so, it would be nice to return the name of the network. I know in currently code path, the network name is also input, which is the same if we return it. But it serves a confirmation and might be in the future the values are different, eg we attach the cluster name in the prefix.

Reply:
As said in another PR, the SDK cannot rely on any in-memory state, because that state would get lost if the process is interrupted and restarted. This happens all the time when you use the CLI: first a process is started to create the network, then a new process is started to run a node, etc. So returning some identifier from this method would serve no purpose, because the returned value would be erased shortly after. The SDK needs to either compute the names of the resources it needs to interact with, or it needs to discover them on the fly.

The only alternative would be to store all the information in a file, but that would be excessive: Docker already tracks all the state for us, so there's no reason to duplicate that state. If we did, then we would run into issues related to state decoherence.

What we're doing here with Docker by creating networks with a fixed name, or creating containers with a fixed prefix, is not different from the way a filesystem works: when you have some software that wants to store some files, those files are likely placed into `/var/my-application/my-file`. Here, instead of `/var/my-application` we have `cluster-name`, instead of `/` we have `_`, and instead of `my-file` we have `container-name`. Instead of querying the filesystem, we query Docker. We generally don't need to keep track of what files we placed in a filesystem, because the filesystem provides this functionality for us; here we don't need to keep track of what we put into Docker, because Docker keeps track of it for us. As long as we use a predictable naming convention (in filesystem terms: a predictable path), then we can discover and manage our resources easily.

Reply:
I am not saying storing the network name, but just return it as the function response. This serves as a confirmation to user that network is created with the expected name. But current implementation also works, and works for the near term goal. It's okay if you keep the interface.